scylla/README.md

```diff
@@ -0,0 +1,96 @@
+<!--
+Copyright (c) 2015 YCSB contributors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you
+may not use this file except in compliance with the License. You
+may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License. See accompanying
+LICENSE file.
+-->
+
+# Scylla CQL binding
+
+Binding for [Scylla](https://www.scylladb.com/), using the CQL API
+via the [Scylla
+driver](https://github.com/scylladb/java-driver/).
+
+## Creating a table for use with YCSB
+
+For keyspace `ycsb`, table `usertable`:
+
+    cqlsh> create keyspace ycsb
+        WITH REPLICATION = {'class' : 'SimpleStrategy', 'replication_factor': 3 };
+    cqlsh> USE ycsb;
+    cqlsh> create table usertable (
+        y_id varchar primary key,
+        field0 varchar,
+        field1 varchar,
+        field2 varchar,
+        field3 varchar,
+        field4 varchar,
+        field5 varchar,
+        field6 varchar,
+        field7 varchar,
+        field8 varchar,
+        field9 varchar);
+
+**Note that `replication_factor` and consistency levels (below) will affect performance.**
+
+## scylla Configuration Parameters
+
+- `hosts` (**required**)
+  - scylla nodes to connect to.
+  - No default.
+
+* `port`
+  * CQL port for communicating with scylla cluster.
+  * Default is `9042`.
+
+- `scylla.keyspace`
+  Keyspace name - must match the keyspace for the table created (see above).
+  See http://docs.datastax.com/en/cql/3.1/cql/cql_reference/create_keyspace_r.html for details.
+
+  - Default value is `ycsb`
+
+- `scylla.username`
+- `scylla.password`
+  - Optional user name and password for authentication. See http://docs.datastax.com/en/scylla/2.0/scylla/security/security_config_native_authenticate_t.html for details.
+
+* `scylla.readconsistencylevel`
+* `scylla.writeconsistencylevel`
+
+  * Default value is `ONE`
+  - Consistency level for reads and writes, respectively. See the [DataStax documentation](http://docs.datastax.com/en/scylla/2.0/scylla/dml/dml_config_consistency_c.html) for details.
+  * *Note that the default setting does not provide durability in the face of node failure. Changing this setting will affect observed performance.* See also `replication_factor`, above.
```

Comment:
ONE is default tho https://github.com/scylladb/java-driver/blob/3.0.1-scylla/driver-core/src/main/java/com/datastax/driver/core/QueryOptions.java#L28

Reply:
I understand that the default an application gets if it has no opinions is ONE. I'm letting you know that the YCSB application default is strong consistency.

Reply:
I can replace the consistency level that it uses with QUORUM. However, it will not make consistency guarantees stronger. It will stay Eventually Consistency as it were. QUORUM only means that the probability of getting stale read is smaller. Another thing is that with writes this CL thing does not reduce the number of replicas data gets written to - it only allows not to wait for the replication. CL=QUORUM for RF = 3 will be 2. CL=ALL for RF = 3 will be 3. Even with CL=ALL reads, the consistency is not getting stronger. Cassandra binding also uses ONE.

Do you want me to change it to QUORUM?

Reply:
yes please, quorum. if you'd like to submit a PR to correct the cassandra binding that would be great.

Reply:
done

Reply:
https://github.com/brianfrankcooper/YCSB/pull/1487 for Cassandra